opslevel/resource_opslevel_group.go

```diff
@@ -0,0 +1,189 @@
+			},
+			"teams": {
+				Type:        schema.TypeList,
+				Description: "The teams where this group is the direct parent.",
+				ForceNew:    false,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceGroupCreate(d *schema.ResourceData, client *opslevel.Client) error {
+	members := []opslevel.MemberInput{}
+	for _, member := range getStringArray(d, "members") {
+		members = append(members, opslevel.MemberInput{Email: member})
+	}
+
+	teams := []opslevel.IdentifierInput{}
+	for _, team := range getStringArray(d, "teams") {
+		if opslevel.IsID(team) {
+			teams = append(teams, opslevel.IdentifierInput{Id: team})
+		} else {
+			teams = append(teams, opslevel.IdentifierInput{Alias: graphql.String(team)})
+		}
+	}
+
+	input := opslevel.GroupInput{
+		Name:        d.Get("name").(string),
+		Description: d.Get("description").(string),
+		Parent:      opslevel.NewIdentifier(d.Get("parent").(string)),
+		Members:     &members,
+		Teams:       &teams,
+	}
+	resource, err := client.CreateGroup(input)
+	if err != nil {
+		return err
+	}
+	d.SetId(resource.Id.(string))
+
+	return resourceGroupRead(d, client)
+}
+
+func resourceGroupRead(d *schema.ResourceData, client *opslevel.Client) error {
+	id := d.Id()
+
+	resource, err := client.GetGroup(id)
+	if err != nil {
+		return err
+	}
+
+	groupMembers, err := resource.Members(client)
+	if err != nil {
+		return err
+	}
+	members := flattenMembersArray(groupMembers)
+
+	descendantTeams, err := resource.DescendantTeams(client)
+	if err != nil {
+		return err
+	}
+	teams := flattenTeamsArray(descendantTeams)
+
+	if err := d.Set("alias", resource.Alias); err != nil {
+		return err
+	}
+	if err := d.Set("name", resource.Name); err != nil {
+		return err
+	}
+	if err := d.Set("description", resource.Description); err != nil {
+		return err
+	}
+	if err := d.Set("parent", resource.Parent.Alias); err != nil {
+		return err
+	}
+	if err := d.Set("members", members); err != nil {
+		return err
+	}
+	if err := d.Set("teams", teams); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceGroupUpdate(d *schema.ResourceData, client *opslevel.Client) error {
+	input := opslevel.GroupInput{}
+
+	members := []opslevel.MemberInput{}
+	for _, member := range getStringArray(d, "members") {
+		members = append(members, opslevel.MemberInput{Email: member})
+	}
+
+	teams := []opslevel.IdentifierInput{}
+	for _, team := range getStringArray(d, "teams") {
+		if opslevel.IsID(team) {
+			teams = append(teams, opslevel.IdentifierInput{Id: team})
+		} else {
+			teams = append(teams, opslevel.IdentifierInput{Alias: graphql.String(team)})
+		}
+	}
+
+	if d.HasChange("name") {
+		input.Name = d.Get("name").(string)
+	}
+	if d.HasChange("description") {
+		input.Description = d.Get("description").(string)
+	}
+	if d.HasChange("parent") {
+		input.Parent = opslevel.NewIdentifier(d.Get("parent").(string))
+	}
+	if d.HasChange("members") {
```

Comment:
We should test to make sure that if we don't go into this block the "update" call doesn't wipe the current members.

Same for Teams.

We may need to do some shenanigans here

Reply:
Pretty sure this is working as expected since it wasn't wiping them out when I'd run a subsequent apply with no changes to this field.  If it was wiping them out, I would expect to see them cleared out and then try to add them back in on another apply.  I ran a few applies in a row and it never tried to add them back in.